### ISSUE TITLE
Trainer args consistency

### ISSUE THREAD

**Issue by rasbt** [Lightning-AI/lit-gpt#951]
For consistency, should it be `io.destination_path` (or perhaps even better `io.out_dir`) in 

```bash
python scripts/prepare_alpaca.py \
   --destination_path data/alpaca \
   --checkpoint_dir checkpoints/google/gemma-2b


python finetune/lora.py \
   --io.train_data_dir data/alpaca \
   --checkpoint_dir checkpoints/google/gemma-2b
```

**Comment by carmocca** [comment]
@awaelchli Is refactoring the current `prepare_*.py` scripts in scope for #950?

**Comment by awaelchli** [comment]
Yes, with https://github.com/Lightning-AI/lit-gpt/pull/950 the goal is to remove these prepare scripts (with the exception of the pretraining datasets which need to be optimized in advance). We haven't made any decisions yet and I'd like to discuss it, but my vision was that the scripts get a datamodule as input which encapsulate all data related args.

**Comment by carmocca** [comment]
Cool. In that case https://github.com/Lightning-AI/lit-gpt/issues/954 replaces this issue
